src/s3/client.rs

```diff
@@ -456,7 +456,18 @@ impl<'a> Client<'a> {
                 .build_url(&method, region, query_params, bucket_name, object_name)?;
         self.build_headers(headers, query_params, region, &url, &method, body);
 
-        let client = reqwest::Client::new();
+        let client;
+        if object_name.unwrap_or_default().to_string().is_empty() && method == Method::GET {
+            client = reqwest::Client::builder()
+                .no_gzip() // needed to ensure no automatic decompression on GetObject
+                .danger_accept_invalid_certs(self.ignore_cert_check)
+                .build()?;
```

Comment:
create a client with `.no_gzip()` and `.danger_accept_invalid_certs(self.ignore_cert_check)` irrespective of object name availability.